pallets/liquid-staking/src/lib.rs

```diff
@@ -889,6 +894,56 @@ pub mod pallet {
 
             Ok(().into())
         }
+
+        /// Cancel unstake
+        #[pallet::weight(10_000)]
+        #[transactional]
+        pub fn cancel_unstake(
+            origin: OriginFor<T>,
+            #[pallet::compact] amount: BalanceOf<T>,
+        ) -> DispatchResultWithPostInfo {
+            let who = ensure_signed(origin)?;
+
+            Unlockings::<T>::try_mutate(&who, |b| -> DispatchResultWithPostInfo {
+                let chunks = b.as_mut().ok_or(Error::<T>::NoUnlockings)?;
+                let mut cancelled: Balance = Zero::zero();
+
+                while let Some(last) = chunks.last_mut() {
+                    if last.era != Self::current_era() + T::BondingDuration::get() + 1
+                        || cancelled >= amount
+                    {
+                        break;
+                    }
```

Comment:
This may be always `break`, because `last.era` is calculated by `let target_era = Self::current_era() + T::BondingDuration::get() + 1;` however, this two `Self::current_era()` is different.

Reply:
the current `cancel_unstake` only works when users are cancelling in the same era. Because when we advance era, we will start to go to relaychain to bond

Reply:
this may make users confused if we only allow them cancel_unstake in the same era, there are actually 29 eras for an unlocking period, `cancel_unstake` suppose to let them cancel in any era before the last.
but if they found that it doesn't work during era 2~29 and only works in era1. this doesn't make sense.

Reply:
yes but feel that should be `fast_unstake` domain if we would like to implement feature like that